test/Test/Util/RawPage.hs

```diff
@@ -22,12 +24,26 @@ import           Test.Tasty.HUnit (Assertion, assertFailure)
 import           Test.Tasty.QuickCheck (Property, counterexample)
 
 -- | Convert prototype 'PageLogical' to 'RawPage'.
-toRawPage :: PageLogical -> (RawPage, BS.ByteString)
-toRawPage p = (page, sfx)
+toRawPage :: PageLogical -> (RawPage, [RawOverflowPage])
+toRawPage p = (page, overflowPages)
   where
     bs = serialisePage $ encodePage p
     (pfx, sfx) = BS.splitAt 4096 bs -- hardcoded page size.
-    page = case SBS.toShort pfx of SBS.SBS ba -> makeRawPage (ByteArray ba) 0
+    page          = makeRawPageBS pfx
+    overflowPages = [ makeRawOverflowPageBS sfxpg
+                    | sfxpg <- takeWhile (not . BS.null)
+                                 [ BS.drop n sfx | n <- [0, 4096 .. ] ]
+                    ]
```

Comment:
This looks like it doesn't satisfy the `makeRawOverflowPage` invariant that the length of `sfxpg` be 4096 or less
```suggestion
    overflowPages = [ makeRawOverflowPageBS sfxpg
                    | BS.take 4096 sfxpg <- takeWhile (not . BS.null)
                                 [ BS.drop n sfx | n <- [0, 4096 .. ] ]
                    ]
```
AFAICS it's not caught in a test -- maybe because the generated pages are too small?

Reply:
I've tweaked the size distribution of the prototype's `LogicalPage` so that it generates bigger pages with the default QC size of 100. Previously it was generating bigger pages but only with size 200+.

And then yes it does detect the bug above, also fixed.